TerrorConsole/Assets/_Main/Scripts/LevelManager/LevelManager.cs

```diff
@@ -13,10 +13,21 @@ public enum LevelState
 
     public partial class LevelManager : Singleton<ILevelManagerSource>, ILevelManagerSource
     {
+
+        [SerializeField] private int levelNumber;
+
         public event Action<LevelState> OnLevelStateChange;
 
         [SerializeField] private LevelState currentLevelState;
 
+        private SaveLevelData _saveLevelData;
+
+        protected override void Awake()
+        {
+            base.Awake();
+            _saveLevelData = SaveSystemManager.Source.GetLoadedGame().GetLevelData(levelNumber);
```

Comment:
Here you are breaking encapsulation a bit, `LevelManager` can be the one calling the Save system but it should not store anything for later use, is like taking control of part of that system.

`LevelManager` should have methos that calls the Save System, but it should not store a reference to `SaveLevelData`